src/uproot/source/fsspec.py

```diff
@@ -13,16 +15,28 @@ class FSSpecSource(uproot.source.chunk.Source):
     to get many chunks in one request.
     """
 
-    def __init__(self, file_path, **kwargs):
+    def __init__(self, file_path, **options):
         import fsspec.core
 
+        default_options = uproot.reading.open.defaults
+        self._use_threads = options.get("use_threads", default_options["use_threads"])
+        self._num_workers = options.get("num_workers", default_options["num_workers"])
+
         # TODO: is timeout always valid?
 
         # Remove uproot-specific options (should be done earlier)
-        exclude_keys = set(uproot.reading.open.defaults.keys())
-        opts = {k: v for k, v in kwargs.items() if k not in exclude_keys}
+        exclude_keys = set(default_options.keys())
+        opts = {k: v for k, v in options.items() if k not in exclude_keys}
 
         self._fs, self._file_path = fsspec.core.url_to_fs(file_path, **opts)
+
+        if self._use_threads and self._fs.__class__.__name__ == "HTTPFileSystem":
```

Comment:
```suggestion
        if self._use_threads and type(self._fs).__name__ == "HTTPFileSystem":
```

I see that we can't do an `isinstance` check against `fsspec.implementations.http.HTTPFileSystem` because we don't know if `fsspec.implementations.http` is installed. It's an optional dependency of fsspec.